### PR TITLE
dynamicconfig: Enables dynamic-config and drift-checker by default

### PR DESCRIPTION
**Pull request by ovidiutirla**
tl;dr
The agent will monitor the designated ConfigMap and automatically synchronize any changes with its in-memory database, StateDB. Create a new cell in the Agent that watches the configured ConfigMap and creates a StateDB table to store the key-value pairs. The Agent cell will expose the read-only version of the StateDB table.
The monitored sources are configured by the initContainer at start-up, see https://github.com/cilium/cilium/blob/236bc2eac4c36cff60e0079f06d33816b3344ccc/pkg/option/resolver/resolver.go#L110

--
**Drift Checker** will compare the DynamicConfig Table content with the Agent config and compute the drift delta, logging the deltas and publish relevant metrics. This functionality allows the cluster administrators to easily track config drifts.
Publishes metrics counting the delta length and labeling by the checksum.

<details>

<summary> Grafana metrics</summary>
Agent start -> Drift appears -> delete all agent pods
image

Start -> Drift -> kubectl rollout restart daemonset cilium -n kube-system

image

</details>

Related: https://github.com/cilium/cilium/issues/27972, https://github.com/cilium/cilium/issues/34101

```release-note
Introduces cilium-config ConfigMap monitoring, allowing the agent to automatically synchronize changes in designated sources with its in-memory database. A new drift checker compares the ingested configurations with the agent's configuration, logging any differences and publishing metrics for easy tracking of configuration drifts. 
```

### CODE REVIEW

**Review by bimmlerd — CHANGES_REQUESTED** [review]
Hi- code changes are trivial and are ok, but the implications aren't clear to me yet
Can you motivate a bit more why this should be enabled by default? Has it seen prod use already? I'm asking because turning it on seems to cause a related panic in tests https://github.com/cilium/cilium/actions/runs/12275147175/job/34249594941#step:13:894 so I'm not sure this is ready.

Re 1.17 - I believe the feature freeze is in effect until 1.17 is branched, so I'm not sure this can still go in unfortunately.

**Comment by ovidiutirla** [comment]
@bimmlerd, coverted to draft until I investigate the failure as it was not failing previously.

**Comment by bimmlerd** [comment]
Can you motivate why this should be enabled by default?

**Comment by ovidiutirla** [comment]
@bimmlerd, yes, forgot about it, I think the main motivation is in #27972. 

But the benefits dynamic-config brings can be used in multiple features and the overhead it brings is low and should be safe, from scalability perspective is not significantly impacting the cluster stability. Basically, we'll want to provide these benefits by default instead of another configuration the end user needs to make in order to get this.

**Review by bimmlerd — APPROVED** [review]

**Review by viktor-kurchenko — APPROVED** [review]
